lib/utils/udpsender.js

```diff
@@ -0,0 +1,20 @@
+require('./KalturaConfig');
+
+var PORT = KalturaConfig.config.udpsender.PORT;
+var HOST = KalturaConfig.config.udpsender.HOST;
+var client = require('dgram').createSocket('udp4');
+
+var sendFunction = function (message,response){
+    var message_buffer = new Buffer(message);
+    response.log('#UDP ' + message_buffer + ' sent to ' + HOST + ':' + PORT);
+    client.send(message_buffer, 0, message_buffer.length,PORT, HOST,
+        function (err, bytes) {
+            if (err) {
+                client.close();
+            }

```

Comment:
The main advantage of having the udp packets which are very light weight is for getting real time alerts not depending on parsing logs etc

Sent from my iPhone

On 14 ????? 2016, at 8:37, nirbe <notifications@github.com<mailto:notifications@github.com>> wrote:


In lib/utils/udpsender.js<https://github.com/kaltura/play-server/pull/244#discussion_r59665344>:

> @@ -0,0 +1,20 @@
> +require('./KalturaConfig');
> +
> +var PORT = KalturaConfig.config.udpsender.PORT;
> +var HOST = KalturaConfig.config.udpsender.HOST;
> +var client = require('dgram').createSocket('udp4');
> +
> +var sendFunction = function (message,response){
> +    var message_buffer = new Buffer(message);
> +    response.log('#UDP ' + message_buffer + ' sent to ' + HOST + ':' + PORT);
> +    client.send(message_buffer, 0, message_buffer.length,PORT, HOST,
> +        function (err, bytes) {
> +            if (err) {
> +                client.close();
> +            }


  1.  this deserve some Error message to be written to logs.
  2.  what happens if you closed the client and but later call to sendFunction - will it work? or send an exception?

As a side note I think it would be safer to have an external process collect logs for ELK instead of send it from the ad-stitching code. This way errors and exceptions won't interfere with the code.

-
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/kaltura/play-server/pull/244/files/5df3080aa6fb29f7523f4380a2c3060f0865ff15#r59665344>